src/main/telemetry/telemetry.c

```diff
@@ -226,3 +232,90 @@ void telemetryProcess(uint32_t currentTime)
 #endif
 }
 #endif
+
+typedef struct sensorTableEntry_s {
+    sensor_e sensor;
+    uint32_t requiredFeatures;
+#ifdef SENSOR_NAMES
+    const char *name;
+#endif
+} sensorTableEntry_t;
+
+#ifdef SENSOR_NAMES
+#define SENSOR_ENTRY(a,b,c) a,b,c
+#else
+#define SENSOR_ENTRY(a,b,c) a,b
+#endif
+
+static const sensorTableEntry_t sensorTable[] = {
+    { SENSOR_ENTRY(ESC_SENSOR_CURRENT, FEATURE_ESC_SENSOR, "ESC_CURRENT") },
+    { SENSOR_ENTRY(ESC_SENSOR_VOLTAGE, FEATURE_ESC_SENSOR, "ESC_VOLTAGE") },
+    { SENSOR_ENTRY(ESC_SENSOR_RPM, FEATURE_ESC_SENSOR, "ESC_RPM") },
+    { SENSOR_ENTRY(ESC_SENSOR_TEMPERATURE, FEATURE_ESC_SENSOR, "ESC_TEMPERATURE") },
+};
+
+#define SENSOR_TABLE_ENTRY_COUNT (sizeof(sensorTable) / sizeof(sensorTableEntry_t))
+
+#define SENSOR_GET_FLAG(sensor) (1 << (sensor - 1))
+
+bool telemetryIsSensorEnabled(sensor_e sensor)
+{
+    return telemetryConfigMutable()->enabledSensors & SENSOR_GET_FLAG(sensor);
+}
+
+void telemetrySensorSet(sensor_e sensor, uint32_t *sensors)
+{
+    *sensors |= SENSOR_GET_FLAG(sensor);
+}
+
+void telemetrySensorClear(sensor_e sensor, uint32_t *sensors)
+{
+    *sensors &= ~(SENSOR_GET_FLAG(sensor));
+}
+
+void telemetrySensorEnable(sensor_e sensor)
+{
+    telemetrySensorSet(sensor, &(telemetryConfigMutable()->enabledSensors));
+}
+
+void telemetrySensorDisable(sensor_e sensor)
+{
+    telemetrySensorClear(sensor, &(telemetryConfigMutable()->enabledSensors));
+}
+
+uint8_t telemetrySensorTableEntryCount(void)
+{
+    return SENSOR_TABLE_ENTRY_COUNT;
+}
+
+sensor_e telemetrySensorForTableIndex(int idx)
+{
+    return (idx >= 0 && idx < (int)SENSOR_TABLE_ENTRY_COUNT) ? sensorTable[idx].sensor : 0;
+}
+
+uint32_t telemetrySensorMaskForTableIndex(int idx)
+{
+    sensor_e sensor = telemetrySensorForTableIndex(idx);
+    if (sensor == 0) {
+        return 0;
+    }
+    return SENSOR_GET_FLAG(sensor);
+}
+
+bool telemetrySensorIsAvailable(int idx)
+{
+    if (idx >= 0 && idx < (int)SENSOR_TABLE_ENTRY_COUNT) {
+        return featureIsEnabled(sensorTable[idx].requiredFeatures | FEATURE_TELEMETRY);
```

Comment:
This tests for any bit set, you want all bits set
(featureIsEnabled expects only one bit set in parameter)

Reply:
```
featureMask() & (sensorTable[idx].requiredFeatures | FEATURE_TELEMETRY) ==  (sensorTable[idx].requiredFeatures | FEATURE_TELEMETRY)
```
(And use local variable : `required = (sensorTable[idx].requiredFeatures | FEATURE_TELEMETRY)`